src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -341,22 +371,26 @@ public function channel(): \AMQPChannel
         return $this->amqpChannel;
     }
 
-    public function queue(): \AMQPQueue
+    private function queue(string $queueName): \AMQPQueue
     {
-        if (null === $this->amqpQueue) {
-            $this->amqpQueue = $this->amqpFactory->createQueue($this->channel());
-            $this->amqpQueue->setName($this->queueConfiguration['name']);
-            $this->amqpQueue->setFlags($this->queueConfiguration['flags'] ?? AMQP_DURABLE);
+        if (!isset($this->amqpQueues[$queueName])) {
+            $queueConfig = $this->queuesConfiguration[$queueName];
 
-            if (isset($this->queueConfiguration['arguments'])) {
-                $this->amqpQueue->setArguments($this->queueConfiguration['arguments']);
+            $amqpQueue = $this->amqpFactory->createQueue($this->channel());
+            $amqpQueue->setName($queueName);
+            $amqpQueue->setFlags($queueConfig['flags'] ?? AMQP_DURABLE);
+
+            if (isset($queueConfig['arguments'])) {
+                $amqpQueue->setArguments($queueConfig['arguments']);
             }
+
+            $this->amqpQueues[$queueName] = $amqpQueue;
         }
 
-        return $this->amqpQueue;
+        return $this->amqpQueues[$queueName];
     }
 
-    public function exchange(): \AMQPExchange
+    private function exchange(): \AMQPExchange
```

Comment:
We need to document this BC break. And we also need to consider if it is really needed